README
NAME
    Apache2::WebApp::Plugin::Memcached - Cache::Memcached module wrapper

SYNOPSIS
      my $obj = $c->plugin('Memcached')->method( ... );     # Apache2::WebApp::Plugin::Memcached->method()

        or

      $c->plugin('Memcached')->method( ... );

DESCRIPTION
    Store persistent data using memcached (memory cache daemon).

    <http://www.danga.com/memcached>

PREREQUISITES
    This package is part of a larger distribution and was NOT intended to be
    used directly. In order for this plugin to work properly, the following
    packages must be installed:

      Apache2::WebApp
      Cache::Memcached
      Params::Validate

INSTALLATION
    From source:

      $ tar xfz Apache2-WebApp-Plugin-Memcached-0.X.X.tar.gz
      $ perl MakeFile.PL PREFIX=~/path/to/custom/dir LIB=~/path/to/custom/lib
      $ make
      $ make test
      $ make install

    Perl one liner using CPAN.pm:

      $ perl -MCPAN -e 'install Apache2::WebApp::Plugin::Memcached'

    Use of CPAN.pm in interactive mode:

      $> perl -MCPAN -e shell
      cpan> install Apache2::WebApp::Plugin:Memcached
      cpan> quit

    Just like the manual installation of Perl modules, the user may need
    root access during this process to insure write permission is allowed
    within the installation directory.

CONFIGURATION
    Unless it already exists, add the following to your projects
    *webapp.conf*

      [memcached]
      servers            = 127.0.0.1:11211, 192.168.1.1:11212, 192.168.1.2:11213
      compress_threshold = 10_000

OBJECT METHODS
  connect
    Initialize a new Memcache session across servers.

      my $memd = $c->plugin('Memcached')->connect({
          memcached_servers            => '127.0.0.1:11211, 192.168.1.1:11212, 192.168.1.2:11213',
          memcached_compress_threshold => 10_000,
          debug                        => 1,
        });

      $memd->set( "foo", "bar" );

SEE ALSO
    Apache2::WebApp, Apache2::WebApp::Plugin, Cache::Memcached

AUTHOR
    Marc S. Brooks, <mbrooks@cpan.org> - <http://mbrooks.info>

COPYRIGHT
    This program is free software; you can redistribute it and/or modify it
    under the same terms as Perl itself.

    See <http://dev.perl.org/licenses/artistic.html>



Changes
Revision history for Perl module Apache2::WebApp::Plugin::Memcached

0.07
    - Migrated project SCM and code repository to Google Project hosting.
    - Updated POD and README artistic license URL in COPYRIGHT clause.
    - Changed each module version so that I can verify that the PAUSE packager/uploader script works as expected.
    - Updated license field in META.yml to fix 'License Unknown' issue on CPAN
    - Updated PREREQ_PM module versions in Makefile.PL
    - Updated module versions in META.yml requires field.


Makefile.PL
use ExtUtils::MakeMaker;

require 5.8.8;

# See lib/ExtUtils/MakeMaker.pm for details of how to influence
# the contents of the Makefile that is written.

use Apache::TestMM qw( clean test );

my @scripts = qw( t/TEST );

# accept the configs from command line
Apache::TestMM::filter_args();

# generate test scripts
Apache::TestMM::generate_script('t/TEST');

WriteMakefile(
    NAME         => 'Apache2::WebApp::Plugin::Memcached',
    VERSION_FROM => 'lib/Apache2/WebApp/Plugin/Memcached.pm', # finds \$VERSION
    AUTHOR       => 'Marc S. Brooks (mbrooks@cpan.org)',
    PREREQ_PM => {
        'Apache::Test'     => 0,
        'Apache2::WebApp'  => 0.38,
        'Cache::Memcached' => 1.25,
        'Params::Validate' => 0,
        'Test::More'       => 0,
    },
    clean => {
        FILES => "@{ clean_files() }",
    }
  );

sub clean_files {
    return [@scripts];
}

# Install build script template dependencies (non-standard way)
sub MY::postamble {
'
install::
	cp -rv usr/share/webapp-toolkit/plugin* /usr/share/webapp-toolkit
';
}

lib/Apache2/WebApp/Plugin/Memcached.pm
#----------------------------------------------------------------------------+
#
#  Apache2::WebApp::Plugin::Memcached - Cache::Memcached module wrapper
#
#  DESCRIPTION
#  Store persistent data using memcached (memory cache daemon).
#
#  AUTHOR
#  Marc S. Brooks <mbrooks@cpan.org>
#
#  This module is free software; you can redistribute it and/or
#  modify it under the same terms as Perl itself.
#
#----------------------------------------------------------------------------+

package Apache2::WebApp::Plugin::Memcached;

use strict;
use base 'Apache2::WebApp::Plugin';
use Cache::Memcached;
use Params::Validate qw( :all );

our $VERSION = 0.07;

#~~~~~~~~~~~~~~~~~~~~~~~~~~[  OBJECT METHODS  ]~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~#

#----------------------------------------------------------------------------+
# connect(\%config)
#
# Initialize a new Memcache session across servers.

sub connect {
    my ( $self, $config )
      = validate_pos( @_,
          { type => OBJECT  },
          { type => HASHREF }
          );

    return new Cache::Memcached {
        servers            => [ split( /\,\s+/, $config->{memcached_servers} ) ],
        compress_threshold => $config->{memcached_compress_threshold} || 10_000,
        debug              => $config->{debug}                        || 0,
      };
}

#~~~~~~~~~~~~~~~~~~~~~~~~~~[  PRIVATE METHODS  ]~~~~~~~~~~~~~~~~~~~~~~~~~~~~~#

#----------------------------------------------------------------------------+
# _init(\%config)
#
# Return a reference of $self to the caller.

sub _init {
    my ( $self, $config) = @_;
    return $self;
}

1;

__END__

=head1 NAME

Apache2::WebApp::Plugin::Memcached - Cache::Memcached module wrapper

=head1 SYNOPSIS

  my $obj = $c->plugin('Memcached')->method( ... );     # Apache2::WebApp::Plugin::Memcached->method()

    or

  $c->plugin('Memcached')->method( ... );

=head1 DESCRIPTION

Store persistent data using memcached (memory cache daemon).

L<http://www.danga.com/memcached>

=head1 PREREQUISITES

This package is part of a larger distribution and was NOT intended to be used 
directly.  In order for this plugin to work properly, the following packages
must be installed:

  Apache2::WebApp
  Cache::Memcached
  Params::Validate

=head1 INSTALLATION

From source:

  $ tar xfz Apache2-WebApp-Plugin-Memcached-0.X.X.tar.gz
  $ perl MakeFile.PL PREFIX=~/path/to/custom/dir LIB=~/path/to/custom/lib
  $ make
  $ make test
  $ make install

Perl one liner using CPAN.pm:

  $ perl -MCPAN -e 'install Apache2::WebApp::Plugin::Memcached'

Use of CPAN.pm in interactive mode:

  $ perl -MCPAN -e shell
  cpan> install Apache2::WebApp::Plugin:Memcached
  cpan> quit

Just like the manual installation of Perl modules, the user may need root access during
this process to insure write permission is allowed within the installation directory.

=head1 CONFIGURATION

Unless it already exists, add the following to your projects I<webapp.conf>

  [memcached]
  servers            = 127.0.0.1:11211, 192.168.1.1:11212, 192.168.1.2:11213
  compress_threshold = 10_000

=head1 OBJECT METHODS

=head2 connect

Initialize a new Memcache session across servers.

  my $memd = $c->plugin('Memcached')->connect({
      memcached_servers            => '127.0.0.1:11211, 192.168.1.1:11212, 192.168.1.2:11213',
      memcached_compress_threshold => 10_000,
      debug                        => 1,
    });

  $memd->set( "foo", "bar" );

=head1 SEE ALSO

L<Apache2::WebApp>, L<Apache2::WebApp::Plugin>, L<Cache::Memcached>

=head1 AUTHOR

Marc S. Brooks, E<lt>mbrooks@cpan.orgE<gt> - L<http://mbrooks.info>

=head1 COPYRIGHT

This program is free software; you can redistribute it and/or modify it
under the same terms as Perl itself.

See L<http://dev.perl.org/licenses/artistic.html>

=cut
